### PR TITLE
fix Issue 10966 - Optimizer generates wrong code with try-catch

### PR DESCRIPTION
**Pull request by WalterBright**
2 line fix.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Stable.

**Comment by WalterBright** [comment]
It's a 7 year old bug.

**Review by ghost — APPROVED** [review]
There is a need for official rules on when it is required to target stable or master. In the meantime let's take this as-is

**Comment by MoonlightSentinel** [comment]
There are official rules ... they just are ignored again and again like this.

**Comment by Geod24** [comment]
> Regression or critical bug fixes should always target the stable branch. Learn more about rebasing to stable or the D release process.

I mean the bug even write it. EVERY. TIME.

**Comment by NilsLankila** [comment]
"should" VS "must" or "have to"

**Comment by NilsLankila** [comment]
"it's mandatory for a .... to target the stable branch".

**Comment by wilzbach** [comment]
> There is a need for official rules on when it is required to target stable or master. In the meantime let's take this as-is

There are official rules - https://wiki.dlang.org/DIP75#Branching_strategy
It's right in there:

# TL;DR Safe fixes should always go into stable, all the time. 

That's why there are so many ⚠️⚠️⚠️ in the automated bot message?

We stopped merging Walter's PRs in the past until they are targeting stable and that actually worked for a while.
It's literally just a one-line command to rebase to `stable` and the bot even tells one how to.

**Comment by NilsLankila** [comment]
as were are collaborating on a compiler people have to take care of the semantics. "should" denotes something optional. It does not constitute a rule that a community have to follow. "should" is a gentle a polite invitation but leaves the door opened.

**Comment by NilsLankila** [comment]
let's discuss and put it in the contribution guidlines.

**Comment by WalterBright** [comment]
Changes always come with a risk, meaning "stable" can become destabilized. With ancient bugs, there isn't a critical need to put the fix into stable, making me reluctant to risk the instability. Especially things like changing the optimizer, which always has risk to it.
